### ISSUE TITLE
Non standard 4xx sets span status to codes.Error

### ISSUE THREAD

**Issue by scorpionknifes** [open-telemetry/opentelemetry-go#4134]
### Description

A http request for a non standard http status code 4xx error (e.g. 499), sets the span status to `codes.Errors`.
`validateHTTPStatusCode` causes this issue in http middlewares (e.g. [otelecho](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/instrumentation/github.com/labstack/echo/otelecho/v0.42.0/instrumentation/github.com/labstack/echo/otelecho/echo.go#L100)).

`validateHTTPStatusCode` returns `codes.Errors` instead of `codes.Unset` although the http status is in the 4xx range.
Since 499 is not part of the [validRangesPerCategory](https://github.com/open-telemetry/opentelemetry-go/blob/main/semconv/internal/v2/http.go#L356)
It returns `codes.Errors` and `Invalid HTTP status code`

validateHTTPStatusCode - https://github.com/open-telemetry/opentelemetry-go/blob/main/semconv/internal/v2/http.go#L386-L399

### Steps To Reproduce

1. Use any middleware relying on  `validateHTTPStatusCode` e.g. [otelecho](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/instrumentation/github.com/labstack/echo/otelecho/v0.42.0/instrumentation/github.com/labstack/echo/otelecho/echo.go#L100)
2. Call a endpoint that results in a non standard http status code e.g 499
3. span status = `codes.Error`

### Expected behavior

Error codes such as 499 is in range of 4xx and error should be `codes.Unset` instead of `codes.Error`

Alternatively configure it to allow user defined 4xx error codes such as 499?

**Comment by scarshad** [comment]
```go
if category > 0 && category < 4 {
		return codes.Unset, true
	}
```
link: https://github.com/open-telemetry/opentelemetry-go/blob/46f2ce5ca6adaa264c37cdbba251c9184a06ed7f/semconv/internal/v2/http.go#L400-L402

499 is the error status code, and in the error category.

why should it be in category of `codes.Unset`?

**Comment by scorpionknifes** [comment]
> https://github.com/open-telemetry/opentelemetry-go/blob/46f2ce5ca6adaa264c37cdbba251c9184a06ed7f/semconv/internal/v2/http.go#L400-L402
> 
> 499 is the error status code, and in the error category.
> 
> why should it be in category of `codes.Unset`?

It returns a couple lines ahead at https://github.com/open-telemetry/opentelemetry-go/blob/46f2ce5ca6adaa264c37cdbba251c9184a06ed7f/semconv/internal/v2/http.go#L397-L399

resulting in `codes.Error`, I expect it to be `codes.Unset`

**Comment by pellared** [comment]
I agree that it looks like a bug. Reference: https://github.com/open-telemetry/opentelemetry-specification/blob/v1.21.0/specification/trace/semantic_conventions/http.md#status. This e.g. how OTel .NET handles it https://github.com/open-telemetry/opentelemetry-dotnet/blob/eb12f9b386f0f3231bd9f41b34b65cc927831e1c/src/OpenTelemetry.Api/Internal/SpanHelper.cs

I can look at it once https://github.com/open-telemetry/opentelemetry-go-contrib/pull/3817 is merged.
